main/index.jsx

```diff
@@ -10,8 +12,10 @@ const Main = ({ children }) => {
   // TODO: Create App Layout
   return (
     <GlobalProvider store={store.current}>
-      <Head />
-      <Layout>{children}</Layout>
+      <MDXProvider components={design}>
```

Comment:
This is not correct even if the component names overlap. MDX components are a set of components to render as markdown. Usually, you need a special logic for it but even if you don't, it's better to create an explicit configuration object with the only components you need to override.

[Check the docs](https://mdxjs.com/table-of-components) to find the full list of available options.

Reply:
I am not sure if I am following. I am not trying to override anything. I am allowing to use design components in mdx files. For example, the <Link /> may be used for redirecting to FAQ.

Reply:
I am afraid you don't do so. This is the configuration object for the markdown components, which are rendered when the markdown is originally transpiled. You can clearly see the configuration in the last example on the [page about components configuration (advanced one)](https://mdxjs.com/advanced/components). Also, from the page I sent:

> MDXProvider uses React Context to provide the component mapping internally to MDX when it renders. The following components are rendered from Markdown, so **these can be keys** in the component object you pass to MDXProvider.

Of course, you **can** pass there anything but there are a limited set of keys you **shall** pass. It's fine if it works in a way you expected but I would rather pass pure configuration with an explicit components remapping.

If you do it without any condition like it is now any further `@oacore/design` update may create a bug in this place which will be very tricky to locate.

Reply:
You are right that it can introduce a bug in the future. On the other hand all of our design components starting with capital character. 

> Of course, you can pass there anything but there are a limited set of keys you shall pass.

there is not a limited set of keys which should be passed. The defaults are merged with new ones I think (tested). So I can use something like this and it gets rendered correctly:

```
# h1 title
We <Link href='/' external>found a link</Link>.
```

---
Anyway, I will remove design components since there are not used yet. In the future, I will add just used ones. Something like:

```
<MDXProvider components={{Link: design.Link}>...</MDXProvider>
```